uint32 rate
---
bool succeed